### DISCUSSION TITLE
Render state diagrams similar to how sequence diagrams are rendered (left to right, top to bottom)

### DISCUSSION THREAD

**Discussion by simplytech** [plantuml/plantuml#1180] [Ideas]
_I love PlantUML and I use it a lot to quickly and efficiently draw diagrams. A big thanks to all the contributors!_

When a State Machine becomes quite complex, the automatic layout of GraphViz doesn't always give an understandable diagram. Moreover, as I'm adding states and transitions, GraphViz does **major** shuffling and reorganization (sometimes even after adding one tiny transition). This makes it difficult to stay focused on the task at hand. While PlantUML provides some tweaks such as `skinParam lineType ortho` which improve the situation a bit, I'm wondering if the following idea might be possible:

**Render state diagrams similar to how sequence diagrams are rendered (left to right, top to bottom).**

When I used to draw State Diagrams by pencil and paper, I used to prefer to draw the states on top of the page, left to right, with a vertical line under each state. For drawing a transition, I draw an arrow from one vertical line to another. For better visuals, I draw a large dot at the start of the arrow (since the arrow might cross several lines before reaching the destination).

From the docs, it seems that a sequence diagram is not rendered using GraphViz. Therefore, given the nature of sequence diagrams (i.e. that they are ordered in chronological sequence top to bottom), it might be easier for developers of PlantUML to draw these diagrams without GraphViz. So, I'm wondering if the developers might be interested in providing an option to render a state diagram using the same rendering engine that is used to render sequence diagrams.

**What I've tried so far:**

Since I can interpret a diagram in anyway I want, I did try to simply use a sequence diagram for my state machine and just name the _states_ as _participants_, the _transitions_ as _messages_, and `hide footbox`. However, I was not able to achieve the equivalent of _sub-states_ (although I can group _participants_ in _boxes_) because I need to transition from _sub-state_ to _state_ (and vice versa) but I cannot have a _message_ from a _box_ to another _box_, or from a _box_ to a _participant_ (which makes sense for a sequence diagram, of course).

Please let me know your thoughts.

**Comment by arnaudroques** [comment]
> 
> Therefore, given the nature of sequence diagrams (i.e. that they are ordered in chronological sequence top to bottom), it might be easier for developers of PlantUML to draw these diagrams without GraphViz. So, I'm wondering if the developers might be interested in providing an option to render a state diagram using the same rendering engine that is used to render sequence diagrams.
> 

Yes, that's a good idea and a good suggestion.

Now, our main issue is that our TODO list already too long, so we won't be able to work on this for now.
Sorry about that!

But we keep the idea :-)
